Lesson18/main.js

```diff
@@ -0,0 +1,213 @@
+const slideList = document.getElementById("js-slid-list");
+const slideWrapper = document.getElementById("js-slide-wrapper");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+}
+
+const createLoading = () => {
+  const loading = createElementWithClassName("img", "loading");
+  loading.id = "js-loading";
+  loading.src = "./loading-circle.gif";
+  return loading;
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const createErrorMessage = (error) => {
+  const errorMessage = createElementWithClassName("p", "error-message");
+  errorMessage.textContent = error;
+  return errorMessage;
+}
+
+const fetchErrorHandling = async (response) => {
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error("サーバーエラーが発生しました");
+  }
+}
+
+const getJsonOrError = async (url) => {
+  const response = await fetch(url);
+  const json = await fetchErrorHandling(response);
+  return json;
+}
```

Comment:
I learned a lot from the following when I got the image and how to handle the error.

Reply:
Thank you very much😄.